arm9/lib/src/OS_reset.c

```diff
@@ -136,7 +141,48 @@ enum
     CARD_ENUM_END
 };
 
-void OSi_ReadCardRom32(u32 src, void *dst, s32 len) //should be static, can't mark as such
+
+ARM_FUNC void OSi_ReloadRomData(void)
+{
+    u32 header = (u32)HW_ROM_HEADER_BUF;
+    const u32 rom_base = *(u32 *)HW_ROM_BASE_OFFSET_BUF;
+
+    if (rom_base >= 0x8000)
+    {
+        OSi_ReadCardRom32(rom_base, (void *)header, 0x160);
+    }
+
+    u32 src_arm9 = *(u32 *)(header + 0x20);
```

Comment:
Maybe we should have a struct for rom header?

Reply:
maybe, but unsure if that would match, since this part of the code has some weird assembly generated by it